File: tutorials/python/loading_balance/lb_worker/piworker.py
from consts import piConst
from spa.clientside import CAsyncServiceHandler as CAHandler, \
    CScopeUQueue as Csb


class PiWorker(CAHandler):
    def __init__(self):
        super(PiWorker, self).__init__(piConst.sidPiWorker)

    def OnResultReturned(self, req_id, q):
        if self.RouteeRequest and req_id == piConst.idComputePi:
            dStart = q.LoadDouble()
            dStep = q.LoadDouble()
            nNum = q.LoadUInt()
            dX = dStart + dStep / 2
            dd = dStep * 4.0
            ComputeRtn = 0.0
            for n in range(0, nNum):
                dX += dStep
                ComputeRtn += dd / (1 + dX * dX)
            q.SetSize(0)
            self.SendRouteeResult(Csb().SaveDouble(ComputeRtn))

File: tutorials/csharp/loading_balance/lb_worker/piworker.cs
using SocketProAdapter;
using SocketProAdapter.ClientSide;

public class PiWorker : CAsyncServiceHandler
{
	public PiWorker() : base(piConst.sidPiWorker)
	{
	}

    protected override void OnResultReturned(ushort sRequestId, CUQueue UQueue)
    {
        if (RouteeRequest)
        {
            switch (sRequestId)
            {
                case piConst.idComputePi:
                    {
                        double dStart;
                        double dStep;
                        int nNum;
                        UQueue.Load(out dStart).Load(out dStep).Load(out nNum);
                        double dX = dStart + dStep / 2;
                        double dd = dStep * 4.0;
                        double ComputeRtn = 0.0;
                        for (int n = 0; n < nNum; n++)
                        {
                            dX += dStep;
                            ComputeRtn += dd / (1 + dX * dX);
                        }
                        SendRouteeResult(ComputeRtn, dStart);
                    } 
                    break;
                default:
                    break;
            }
        }
    }
}
